canonical-web-and-design/cn.ubuntu.com
# Update `export-page` to create DB migrations

It's much neater if the database can be brought up-to-date with a simple
standard `./manage.py migrate` command. This is already run by existing
deployment solutions.

To achieve this, we need to have the new page data be imported as
[django migrations](https://docs.djangoproject.com/en/1.9/topics/migrations/).

Unfortunately, django will only run a migration once per database, so
we have to create a new migration for each page.

To this end I've done the following:
- Added a migration template (`page.py.jinja2`)
- Add a new `manage.py` script, `create-page-migration`
- Updated `make export-page` to run `create-page-migration` as well
- Remove `make import-new-pages` as it's now redundant. Run `make db-update` or `./manage.py migrate` instead
# QA

Start with just one page in your CMS, that isn't `desktop`.                                                                                                                                   
- `make export-page` - This will create .json file and migration file  
- `make db-reset` - This deletes database and reprovisions. Watch your data being imported.
- `make run` - Check the data isn't imported this time.  

Now go to your CMS and check the page has been recreated just as before. It should also have created the `desktop` page.
